SU2_DEF/src/meson.build

```diff
@@ -1,38 +1,40 @@
-                                        'output/filewriter/CFVMDataSorter.cpp',
-                                        'output/filewriter/CFEMDataSorter.cpp',
-                                        'output/filewriter/CSurfaceFEMDataSorter.cpp',
-                                        'output/filewriter/CSurfaceFVMDataSorter.cpp',
-                                        'output/filewriter/CParallelFileWriter.cpp',
-                                        'output/filewriter/CParaviewFileWriter.cpp',
-                                        'output/filewriter/CParaviewBinaryFileWriter.cpp',
-                                        'output/filewriter/CTecplotFileWriter.cpp',
-                                        'output/filewriter/CTecplotBinaryFileWriter.cpp',
-                                        'output/filewriter/CCSVFileWriter.cpp',
-                                        'output/filewriter/CSTLFileWriter.cpp',
-                                        'output/filewriter/CSU2FileWriter.cpp',
-                                        'output/filewriter/CSU2BinaryFileWriter.cpp',
-                                        'output/filewriter/CParaviewXMLFileWriter.cpp',
-                                        'output/filewriter/CParaviewVTMFileWriter.cpp',
-                                        'output/filewriter/CSU2MeshFileWriter.cpp',
-                                        'limiters/CLimiterDetails.cpp'])
+su2_cfd_obj = su2_cfd_lib.extract_all_objects()
+
+  su2_def_lib = static_library('SU2core_DEF',
+                               su2_def_src,
+                               install : false,
+                               dependencies : [su2_deps, common_dep],
+                               cpp_args:  [default_warning_flags, su2_cpp_args])
+
+  su2_def_dep = declare_dependency(link_with: su2_def_lib,
+                                   include_directories: su2_def_include)
 
   su2_def = executable('SU2_DEF',
-                        su2_def_src, 
+                        su2_def_src,
                         install: true,
-                        dependencies: [su2_deps, common_dep], 
+                        dependencies: [su2_deps, common_dep],
                         objects : su2_cfd_obj,
                         cpp_args :[default_warning_flags, su2_cpp_args])
 endif
+
+if get_option('enable-autodiff')
+su2_cfd_obj_ad = su2_cfd_lib_ad.extract_all_objects()
+
+  su2_def_lib_ad = static_library('SU2core_DEF_AD',
+                                  su2_def_src,
+                                  install : false,
+                                  dependencies : [su2_deps, codi_dep, commonAD_dep],
+                                  cpp_args:  [default_warning_flags, su2_cpp_args, codi_rev_args])
+
+  su2_def_dep_ad = declare_dependency(link_with: su2_def_lib_ad,
+                                     include_directories: su2_def_include)
+
+endif
```

Comment:
Is this needed?

Reply:
Can you please elaborate on which parts are redundant or need to be addressed?

Reply:
There is no SU2_DEF_AD, so what is this "def_lib_ad" for? Is it SU2_DOT_AD but renamed?